torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -458,3 +493,69 @@ def transpose_matmul(A: torch.Tensor, B: torch.Tensor, Atrans: bool, Btrans: boo
     if Btrans:
         B = B.transpose(-1, -2)
     return torch.matmul(A, B)
+
+
+_pointwise_ops = [
+    operator.add,
+    operator.sub,
+    operator.mul,
+    operator.iadd,
+    operator.isub,
+    operator.imul,
+    operator.itruediv,
+    operator.ifloordiv,
+]
+
+
+@register_graph_pattern(
+    CallFunction(
+        operator.setitem,
+        Arg(),
+        Arg(),
+        CallFunction(
+            _pointwise_ops,
+            CallFunction(
+                operator.getitem,
+                Arg(),
+                Arg(),
+            ),
+            Arg(),
+        ),
+    ),
+    pass_dict=split_cat_pass,
+)
+def boolean_mask_rewritten(match: Match, tensor1, index1, tensor2, index2, const_value):
+    if (
+        isinstance(index1, torch.fx.node.Node)
+        and "example_value" in index1.meta
+        and isinstance(index1.meta["example_value"], FakeTensor)
+        and index1.meta["example_value"].dtype == torch.bool
+        and tensor1 is tensor2
+        and index1 is index2
+        and isinstance(const_value, numbers.Number)
+    ):
+        graph = match.graph
+        node = match.output_node()
+        with graph.inserting_before(node):
+            true_output = graph.call_function(
+                node.args[2].target,
+                args=(tensor1, const_value),
+            )
+            where = graph.call_function(
+                torch.where,
+                args=(index1, true_output, tensor1),
+                kwargs={"out": tensor1},
```

Comment:
hum, I hope the in-place where works lol

Reply:
Will add unit test for this.